airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresQueryUtils.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.postgres;
+
+/**
+ * Utility class to define constants related to querying postgres
+ */
+public class PostgresQueryUtils {
+
+  public static final String NULL_CURSOR_VALUE_WITH_SCHEMA_QUERY =
+      """
+        SELECT
+          (EXISTS (SELECT FROM information_schema.columns WHERE table_schema = '%s' AND table_name = '%s' AND is_nullable = 'YES' AND column_name = '%s'))
+        AND
+          (EXISTS (SELECT from \"%s\".\"%s\" where \"%s\" IS NULL LIMIT 1)) AS %s
+      """;
+  public static final String NULL_CURSOR_VALUE_NO_SCHEMA_QUERY =
+      """
+      SELECT
+        (EXISTS (SELECT FROM information_schema.columns WHERE table_name = '%s' AND is_nullable = 'YES' AND column_name = '%s'))
+      AND
+        (EXISTS (SELECT from \"%s\" where \"%s\" IS NULL LIMIT 1)) AS %s
+      """;
+
+  public static final String TABLE_ESTIMATE_QUERY =
+      """
+            SELECT (SELECT COUNT(*) FROM %s) AS %s,
+            pg_relation_size('%s') AS %s;
```

Comment:
The PR description says the intention is to only estimate table sizes using `pg_relation_size`, but won't this query end up doing a full count also? I believe this is what's causing my sync to hang for minutes (and maybe more) as attempting a count on my 186M row table takes quite a while.

Reply:
If the number of rows is needed, I'd recommend using an estimate via `SELECT reltuples AS estimate FROM pg_class WHERE relname = 'table_name'`. [[1](https://wiki.postgresql.org/wiki/Count_estimate)]

Reply:
aha, you know about this as part of #21499.